### ISSUE TITLE
[tool_crash] _TypeError: (#0      currentPackageConfig (package:flutter_tools/src/dart/package_map.dart:16:56))

### ISSUE THREAD

**Issue by sarathnk** [flutter/flutter#153238]
## Command
```sh
flutter run
```

## Steps to Reproduce
1. ...
2. ...
3. ...

## Logs
_TypeError: (#0      currentPackageConfig (package:flutter_tools/src/dart/package_map.dart:16:56))
```console
#0      currentPackageConfig (package:flutter_tools/src/dart/package_map.dart:16:56)
#1      _loadDwdsDirectory (package:flutter_tools/src/isolated/devfs_web.dart:1236:45)
#2      WebAssetServer.start (package:flutter_tools/src/isolated/devfs_web.dart:270:15)
<asynchronous suspension>
#3      WebDevFS.create (package:flutter_tools/src/isolated/devfs_web.dart:860:22)
<asynchronous suspension>
#4      ResidentWebRunner.run.<anonymous closure> (package:flutter_tools/src/isolated/resident_web_runner.dart:325:19)
<asynchronous suspension>
#5      asyncGuard.<anonymous closure> (package:flutter_tools/src/base/async_guard.dart:111:24)
<asynchronous suspension>
```
```console
[32m[✓][39m Flutter (Channel master, 3.24.0-1.0.pre.528, on macOS 14.5 23F79 darwin-arm64 (Rosetta), locale en-IN)
    [32m•[39m Flutter version 3.24.0-1.0.pre.528 on channel master at /Users/sarathnk/Desktop/development/flutter
    [32m•[39m Upstream repository https://github.com/flutter/flutter.git
    [32m•[39m Framework revision 493c453d57 (2 days ago), 2024-08-09 12:53:15 +0200
    [32m•[39m Engine revision c9ec468c75
    [32m•[39m Dart version 3.6.0 (build 3.6.0-131.0.dev)
    [32m•[39m DevTools version 2.38.0

[32m[✓][39m Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    [32m•[39m Android SDK at /Users/sarathnk/Library/Android/sdk
    [32m•[39m Platform android-34, build-tools 33.0.1
    [32m•[39m Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    [32m•[39m Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    [32m•[39m All Android licenses accepted.

[33m[!][39m Xcode - develop for iOS and macOS (Xcode 15.4)
    [32m•[39m Xcode at /Applications/Xcode.app/Contents/Developer
    [32m•[39m Build 15F31d
    [33m![39m CocoaPods 1.11.2 out of date (1.13.0 is recommended).
        CocoaPods is a package manager for iOS or macOS platform code.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/to/platform-plugins
      To update CocoaPods, see https://guides.cocoapods.org/using/getting-started.html#updating-cocoapods

[32m[✓][39m Chrome - develop for the web
    [32m•[39m Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[32m[✓][39m Android Studio (version 2022.3)
    [32m•[39m Android Studio at /Applications/Android Studio.app/Contents
    [32m•[39m Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    [32m•[39m Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    [32m•[39m Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[32m[✓][39m IntelliJ IDEA Community Edition (version 2024.1)
    [32m•[39m IntelliJ at /Applications/IntelliJ IDEA CE.app
    [32m•[39m Flutter plugin version 81.0.2
    [32m•[39m Dart plugin version 241.17502

[32m[✓][39m Connected device (3 available)
    [32m•[39m macOS (desktop)                 • macos                 • darwin-arm64   • macOS 14.5 23F79 darwin-arm64 (Rosetta)
    [32m•[39m Mac Designed for iPad (desktop) • mac-designed-for-ipad • darwin         • macOS 14.5 23F79 darwin-arm64 (Rosetta)
    [32m•[39m Chrome (web)                    • chrome                • web-javascript • Google Chrome 118.0.5993.117

[32m[✓][39m Network resources
    [32m•[39m All expected network resources are available.

[33m![39m Doctor found issues in 1 category.

```

## Flutter Application Metadata
**Type**: app
**Version**: 1.0.0+1
**Material**: true
**Android X**: false
**Module**: false
**Plugin**: false
**Android package**: null
**iOS bundle identifier**: null
**Creation channel**: master
**Creation framework version**: 493c453d578054912fd7eca655de71ab437e3ce0

**Comment by danagbemava-nc** [comment]
Hi @sarathnk, how often do you get the error? Can we get the full output of `flutter run -v`?

**Comment by mrlegio** [comment]
Hi, I have the same error after updating Flutter to 3.24.0 from 3.22.1
Current version:
```
Flutter 3.24.0 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 80c2e84975 (3 weeks ago) • 2024-07-30 23:06:49 +0700
Engine • revision b8800d88be
Tools • Dart 3.5.0 • DevTools 2.37.2
```

Flutter crash report.
Please report a bug at https://github.com/flutter/flutter/issues.

## command

flutter run -d chrome --web-browser-flag --disable-web-security

## exception

_TypeError: Null check operator used on a null value

```
#0      currentPackageConfig (package:flutter_tools/src/dart/package_map.dart:16:56)
#1      _loadDwdsDirectory (package:flutter_tools/src/isolated/devfs_web.dart:1234:45)
#2      WebAssetServer.start (package:flutter_tools/src/isolated/devfs_web.dart:270:15)
<asynchronous suspension>
#3      WebDevFS.create (package:flutter_tools/src/isolated/devfs_web.dart:860:22)
<asynchronous suspension>
#4      ResidentWebRunner.run.<anonymous closure> (package:flutter_tools/src/isolated/resident_web_runner.dart:328:19)
<asynchronous suspension>
#5      asyncGuard.<anonymous closure> (package:flutter_tools/src/base/async_guard.dart:111:24)
<asynchronous suspension>
```

## flutter doctor

```
[32m[✓][39m Flutter (Channel stable, 3.24.0, on macOS 14.4.1 23E224 darwin-arm64, locale en-US)
    [32m•[39m Flutter version 3.24.0 on channel stable at /Users/paha/StudioProjects/flutter
    [32m•[39m Upstream repository https://github.com/flutter/flutter.git
    [32m•[39m Framework revision 80c2e84975 (3 weeks ago), 2024-07-30 23:06:49 +0700
    [32m•[39m Engine revision b8800d88be
    [32m•[39m Dart version 3.5.0
    [32m•[39m DevTools version 2.37.2

[32m[✓][39m Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    [32m•[39m Android SDK at /Users/paha/Library/Android/sdk
    [32m•[39m Platform android-34, build-tools 34.0.0
    [32m•[39m Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    [32m•[39m Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)
    [32m•[39m All Android licenses accepted.

[32m[✓][39m Xcode - develop for iOS and macOS (Xcode 15.4)
    [32m•[39m Xcode at /Applications/Xcode.app/Contents/Developer
    [32m•[39m Build 15F31d
    [32m•[39m CocoaPods version 1.15.0

[32m[✓][39m Chrome - develop for the web
    [32m•[39m Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[32m[✓][39m Android Studio (version 2024.1)
    [32m•[39m Android Studio at /Applications/Android Studio.app/Contents
    [32m•[39m Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    [32m•[39m Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    [32m•[39m Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)

[32m[✓][39m Connected device (5 available)
    [32m•[39m iPhone Paha (mobile)            • 00008120-0004182611E3C01E            • ios            • iOS 17.5.1 21F90
    [32m•[39m iPhone 13 (mobile)              • 5DA1F4F6-FEBD-4FDB-8C5E-A51B475E3946 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-5 (simulator)
    [32m•[39m macOS (desktop)                 • macos                                • darwin-arm64   • macOS 14.4.1 23E224 darwin-arm64
    [32m•[39m Mac Designed for iPad (desktop) • mac-designed-for-ipad                • darwin         • macOS 14.4.1 23E224 darwin-arm64
    [32m•[39m Chrome (web)                    • chrome                               • web-javascript • Google Chrome 127.0.6533.120

[32m[✓][39m Network resources
    [32m•[39m All expected network resources are available.

[32m•[39m No issues found!
```

**Comment by danagbemava-nc** [comment]
Hi @mrlegio, can we get the full output of `flutter run -d chrome --web-browser-flag --disable-web-security -v`? The output may be huge, so kindly put it into a .txt file and attach it here. 

Thank you

**Comment by danagbemava-nc** [comment]
Thanks for the logs. Labeling this for further investigation.

**Comment by yjbanov** [comment]
Acknowledged.

**Comment by zh4ngx** [comment]
has this been fixed in 3.24.2 or 3.24.3? I see some items about flutter web in the changelog

**Comment by rob-mur** [comment]
Still getting the issue on 3.24.3 so I don't think so.

**Comment by clotodex** [comment]
Also having this issue, so staying on 3.22 for now. If you need any cross-validation or anything let me know. Blocking us from upgrading.

**Comment by rob-mur** [comment]
To provide some extra information here that may help:

- This only happens for me on web, android works ok
- This only happens in debug mode. Profile and release are fine
- @mrlegio mentioned "when I run through Run -> Run, the web starts normally". I assume they may be using android studio or something but at least for me on VsCode 1.94.2 with flutter extension v3.99.20240930 I still get the crash

So to summarise, debugging flutter web on stable doesn't work (at least with my setup). Below is the combined stderr and stdout of `flutter run -d chrome -v`

[flutter run -d chrome -v.txt](https://github.com/user-attachments/files/17659903/flutter.run.-d.chrome.-v.txt)

**Comment by clotodex** [comment]
Interesting observation. For me it does NOT crash if I am in 3.24.4 AND use `--wasm` to launch it.
I dont know if internally some `--release` is implied since in wasm mode you dont get console output right in your terminal.
But just wanted to share for the extra desperate.

**Comment by clotodex** [comment]
I think 3.27 finally fixed this for me - can anyone else confirm?

**Comment by rob-mur** [comment]
Yeah fixed on 27 for me too

**Comment by Pismice** [comment]
> I think 3.27 finally fixed this for me - can anyone else confirm?

still not working on 3.27.3 for me

**Comment by thunze** [comment]
Not sure if this is related, but I got the same crash logs when using Flutter via Nix, and it seems NixOS/nixpkgs#354174 fixed it for me.

**Comment by zh4ngx** [comment]
> Not sure if this is related, but I got the same crash logs when using Flutter via Nix, and it seems [NixOS/nixpkgs#354174](https://github.com/NixOS/nixpkgs/pull/354174) fixed it for me.

Nice find!

**Comment by xprateek** [comment]
####  +1 [Same issue i face in my NixOS]
